tests/test_ramp_fitting.py

```diff
@@ -0,0 +1,408 @@
+import numpy as np
+
+from stcal.ramp_fitting.ramp_fit import ramp_fit_data
+from stcal.ramp_fitting.ramp_fit_class import RampData
+
+DELIM = "-" * 70
+
+# single group intergrations fail in the GLS fitting
+# so, keep the two method test separate and mark GLS test as
+# expected to fail.  Needs fixing, but the fix is not clear
+# to me. [KDG - 19 Dec 2018]
+
+dqflags = {
+    'DO_NOT_USE':       2**0,   # Bad pixel. Do not use.
+    'SATURATED':        2**1,   # Pixel saturated during exposure
+    'JUMP_DET':         2**2,   # Jump detected during exposure
+    'NO_GAIN_VALUE':    2**19,  # Gain cannot be measured
+    'UNRELIABLE_SLOPE': 2**24,  # Slope variance large (i.e., noisy pixel)
+}
+
+
+# -----------------------------------------------------------------------------
+#                           Test Suite
+
+
+def base_neg_med_rates_single_integration():
+    """
+    Ensures negative Poisson
```

Comment:
Not sure that this comment really tells the full story. We don't want negative Poisson noise values, so why would we want to ensure it?  I assume this is meant to indicate something along the lines of "Ensure negative slopes don't result in negative Poisson variance" or something like that?

Reply:
I updated the comments for these functions to make them more clear, especially for the base functions that tests used.